index.js

```diff
@@ -18,39 +18,38 @@ var baseUrl = config.PROD ? 'http://54.183.205.82/alexa/' : 'http://localhost:80
 var applicationId = 'amzn1.ask.skill.7ff009fa-df68-4cd4-b6fd-9500d4791b42';
 
 app.pre = function(req, res, type) {
-  // TODO: check if carvis userId is stored in the session
-  // if no, exchange the amazon userId for carvis userId and store latter in session
-  // move fetch call from app.launch to here
+  //Before req, make sure comes from Alexa skill
   console.log('amazon userId:', req.userId);
   var reqAppId = req.data.session.application.applicationId;
   if (reqAppId !== applicationId && config.PROD) {
     console.log('Invalid applicationId');
     res.fail('Invalid applicationId');
   }
+
+  if(!req.session.carvisUserId) {
+    fetch(baseUrl + 'launch/'+req.userId, {
```

Comment:
the amazon userId is quite long (its not just a few digit #), so perhaps we'd be better off passing it in the request payload rather than as a path parameter? or did you test with realistic amazon userId strings and it worked fine?